cli/pkg/github/git.go

```diff
@@ -0,0 +1,47 @@
+package github
+
+import (
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/go-git/go-git/v5"
+	"github.com/go-git/go-git/v5/plumbing/transport/ssh"
+)
+
+func PublicKey() (*ssh.PublicKeys, error) {
+	var publicKey *ssh.PublicKeys
+	sshPath := os.Getenv("HOME") + "/.ssh/id_ed25519"
+	sshKey, _ := os.ReadFile(sshPath)
+	publicKey, err := ssh.NewPublicKeys("git", sshKey, "")
+	if err != nil {
+		return nil, err
+	}
+	return publicKey, err
+}
+
+func CloneRepo(url string) (string, error) {
+	repoName := strings.TrimSuffix(url[strings.LastIndex(url, "/")+1:], ".git")
+	fmt.Printf("Cloning Repo URL: {%s}\n", url)
+	if strings.HasPrefix(url, "git@github.com") {
```

Comment:
We should not do special handling for github – just get OS credentials and do a clone

Reply:
What if user is not having any credentials and just trying to get public repo? i.e via http? that doesn't required any cred to fetch, just simple clone should work in that case. so i just added handling if projectPath is not `http` i.e starts with `git@`.. then check for cred.

Reply:
If I'm not mistaken, Git URLs with HTTP can also require credentials (ie you can check out a private repo using the HTTP protocol).

In case a repo is public, passing the credentials will just be redundant, but not cause any harm.

Reply:
checking the `endpoint.Protocol` for ssh URL, else it will consider it as `https`